web/src/compose_validate.js

```diff
@@ -243,7 +243,10 @@ function show_wildcard_warnings(stream_id) {
     });
 
     // only show one error for any number of @all or @everyone mentions
-    if ($(`#compose_banners .${CSS.escape(classname)}`).length === 0) {
+    if (
+        $(`#compose_banners .${CSS.escape(classname)}`).length === 0 ||
+        $(`#edit_form_banners .${CSS.escape(classname)}`).length === 0
+    ) {
```

Comment:
This function should just accept a banner container as an argument and do the check based on that specific container.

Reply:
By writing "function", you meant `show_wildcard_warnings` function, right?

Reply:
Reading more carefully, this function can only be called from the compose box right now anyway. 

That said, what I have in mind is that `compose_validate.validate()` would become `compose_validate.validate($textarea)`, and we'd pass that parameter through to these various helpers.

And then I think the only function you'll need is one to, given a composition textarea, to its associated compose banners container -- aka the same thing we need in the `composebox_typeahead` code path and the `uploads` code path. And that can just check if the ID is `compose-textarea`; if it's that, return `compose_banners`, otherwise, aka the message edit case, it can just do `$textarea.closest(".message_edit_form").find(".edit_form_banners")`

I think `get_compose_banner_container($textarea)` would be a very reasonable signature for that function.

Reply:
Looks like `compose_validate.validate()` is only called when the message is being sent, not when you click 'save' after message editing. Therefore, in this code path, $textarea may only be `#compose_banners`.